clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -90,6 +91,8 @@ public void configure(Map<String, ?> configs) throws KafkaException {
         if (clientAuthConfigOverride != null) {
             nextConfigs.put(BrokerSecurityConfigs.SSL_CLIENT_AUTH_CONFIG, clientAuthConfigOverride);
         }
+        nextConfigs.put(SecurityConfig.SECURITY_PROVIDERS_CONFIG,
+                configs.get(SecurityConfig.SECURITY_PROVIDERS_CONFIG));
```

Comment:
Remove about two lines code and something like below? 
copyMapEntries(nextConfigs, configs, SslConfigs.NON_RECONFIGURABLE_CONFIGS);
copyMapEntries(nextConfigs, configs, SslConfigs.RECONFIGURABLE_CONFIGS);
copyMapEntries(nextConfigs, configs, SecurityConfig. SECURITY_PROVIDERS_CONFIG)